connectors/src/connectors/microsoft/index.ts

```diff
@@ -277,39 +280,32 @@ export class MicrosoftConnectorManager extends BaseConnectorManager<null> {
       );
     }
 
-    const client = await getClient(connector.connectionId);
-
     await MicrosoftRootResource.batchDelete({
       resourceIds: Object.keys(permissions),
       connectorId: connector.id,
     });
 
-    const newResourcesBlobs = await concurrentExecutor(
-      Object.entries(permissions).filter(
-        ([, permission]) => permission === "read"
-      ),
-      async ([id]) => {
-        const { nodeType } = typeAndPathFromInternalId(id);
-
-        const { deltaLink } = await getDeltaResults({
-          client,
-          parentInternalId: id,
-          token: "latest",
-        });
-
-        return {
-          connectorId: connector.id,
-          nodeType,
-          internalId: id,
-          currentDeltaLink: deltaLink,
-        };
-      },
-      { concurrency: 5 }
-    );
+    const newResourcesBlobs = Object.keys(permissions)
+      .filter(([, permission]) => permission === "read")
+      .map((id) => ({
+        connectorId: connector.id,
+        nodeType: typeAndPathFromInternalId(id).nodeType,
+        internalId: id,
+      }));
 
     await MicrosoftRootResource.batchMakeNew(newResourcesBlobs);
 
-    const res = await launchMicrosoftFullSyncWorkflow(this.connectorId, null);
+    const nodesToSync = await getSiteNodesToSync(this.connectorId);
+
+    // poupulates deltas for the nodes so that if incremental sync starts before
+    // fullsync populated, there's no error
+    await populateDeltas(this.connectorId, nodesToSync);
+
+    const res = await launchMicrosoftFullSyncWorkflow(
+      this.connectorId,
+      null,
```

Comment:
Why not just removing `fromTs` from `launchMicrosoftFullSyncWorkflow` signature?

Reply:
Good point!